paper_sections/05-computational_and_statistical_issues.Rmd

```diff
@@ -628,9 +636,22 @@ to balance penalties between nested smooth terms (this is why nested random
+To avoid this issue, we recommend both careful choice of basis and setting model degrees of freedom so that group-level terms are either slightly less flexible than the global term or have a smaller null space. In the examples in section III, we used smoothers with an unpenalized null space for the global smoother and ones with no null space for the group-level terms. When using thin plate splines, this can be done with splines with a lower order of derivative penalized in the group-level smoothers than the global smoothers, as lower-order TPRS splines have fewer basis functions in the null space. For example, we used `m=2` (penalizing squared second derivatives) for the global smoother, and `m=1` (penalizing squared first derivatives) for group-level smoothers in models *GS* and *GI*. Another option is to use a lower number of basis functions (`k`) for group-level relative to global terms.  This will reduce the maximum flexibility possible in the group-level terms. We do caution that these are just rules of thumb. 
 
-[^gsnull]: For model *GS* both the factor-smoother, and tensor products of random effect ("re") and other smooth terms do not have a penalized nullspace by construction (they are full rank), as noted above. For model *GI* groupwise terms, we used basis types that had a penalty added to the nullspace, so called "shrinkage" methods: `bs="ts"`, `"cs"`, or `"ps"` have this property.
+As noted above, interpreting the shape of global terms and group-wise deviations seperately for  *GS* models fit using tensor-product group-level terms is complicated by the fact that **mgcv** will drop some basis-functions from the group-level terms to prevent perfect concurvity. 
+For tensor-product smoothers, **mgcv** will generally drop $\le k$ terms from the group-level smoother, where $k$ is the number of basis functions in the global smoother.
+The total number of terms dropped will depend on the smoothers used for the global and group-level terms, and will be dropped essentially at random from different grouping levels. 
```

Comment:
"essentially at random" is this right?

Reply:
This was sloppily worded on my part. "Dropped without respect for grouping level" would be more accurate.

Reply:
`gam.side` will apply the constraint to the the group-level smooth (as it will always (hopefully!) have more variables than the global smoother. However, the basis functions it removes are decided numerically, and will be removed regardless of grouping level (they might all be dropped from one group, or spread out among levels, etc.). I don't think there's much more we can say about it here, without really digging into the guts of `gam.side`.